odl/trafos/wavelet.py

```diff
@@ -652,21 +982,28 @@ def _call(self, x):
             The length of the array depends on the size of input image to
             be transformed and on the chosen wavelet basis.
         """
-        if x.space.ndim == 1:
-            coeff_list = pywt.wavedec(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 2:
-            coeff_list = pywt.wavedec2(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 3:
-            coeff_dict = wavelet_decomposition3d(x, self.wbasis, self.mode,
-                                                 self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_dict, self.size_list)
-
+        if self._axes is None:
+            if x.space.ndim == 1:
+                coeff_list = pywt.wavedec(x, self.wbasis, self.mode,
+                                          self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 2:
+                coeff_list = pywt.wavedec2(x, self.wbasis, self.mode,
+                                           self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 3:
+                coeff_list = wavelet_decomposition3d(x, self.wbasis, self.mode,
+                                                     self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+
+                return self.range.element(coeff_arr)
+        else:
+            coeff_dict = pywt.dwtn(x, self.wbasis, self.mode, self._axes)
```

Comment:
No they are not (though at first one would think that they are). But `wavedec2` using 2 levels returns the coefficients at each level, as for `pywt.dwtn` using `axes = [0, 0, 1, 1]` returns the final resulting coefficients at the "end" level. For an example a 2D case with Daubechies 2 wavelets:

`x = np.ones((32,  32))`
`wbasis = pywt.Wavelet('db2')`
`coeffs_using_wavedec2 = pywt.wavedec2(x, wbasis, mode='sym', level=2)`

returns a list

`[aa0, (ad0, da0, dd0), (ad1, da1, dd1)]`

where  the coefficient in the coarsest level (`aa0, ad0, da0,dd0`) are of size `10 x 10` and the coefficient at the finest (at this case 2nd level, i.e coeffs `ad1, da1, dd1`), are of size `17 x 17`

As for

`coeffs_using_dwtn_with_axes = pywt.dwtn(x, wbasis, mode='sym', axes=[0, 0, 1, 1])`

returns a dictionary with keywords

`['aaaa',  'aaad',  'aada',  'aadd',  'adaa',  'adad',  'adda',  'addd',  'daaa',  'daad',  'dada',  'dadd',  'ddaa',  'ddad',   ddda',  'dddd']`

which all are of (the final size) `10 x 10 `.